### PR TITLE
Adds border styling for homepage search bar on focus

### PR DESCRIPTION
**Pull request by emimuresan**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Documentation Update

## Description
As described in #4663, the homepage search bar should be styled when it's focused in order to better indicate that fact. The example in the issue suggests using a different background color, however i want to propose a solution where we use the border for on focus styling. This i think is more in line with the default browser styling and the styling found in many websites (so it's a pattern familiar to users). I used the same color used in the border for the nav controls. Screenshots below.

## Related Tickets & Documents
#4663 

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
* Default Theme:
Oct-30-2019 22-23-08

* Other Themes:
Screenshot 2019-10-31 at 11 57 48

Screenshot 2019-10-31 at 11 59 28

Screenshot 2019-10-31 at 12 00 15

Screenshot 2019-10-31 at 12 02 16

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

rainbox

### CODE REVIEW

File: app/assets/stylesheets/top-bar.scss [hunk 1]
```diff
@@ -432,6 +431,10 @@
       $black
     );
     -webkit-appearance: none;
+    border: 1px solid transparent;
```
Comment (emimuresan) [inline app/assets/stylesheets/top-bar.scss]: Added in order to avoid a small pop-out effect / reflow

**Comment by emimuresan** [comment]
> This is a great idea! Thanks for the PR @emimuresan! 👏
> 
> How do you think this looks with other themes? Should we usea theme variable for the color?

Hi @jacobherrington ! I wasn't sure if i should make this themable or not. 
If it should be themable, we could either create a new theme variable: `theme-top-bar-search-border`, i'm not sure if there is an existing one that fits.

**Comment by rhymes** [comment]
@emimuresan could you provide a video for all the themes? so that we could evaluate how that looks in all of them. That would be great!

**Comment by emimuresan** [comment]
> @emimuresan could you provide a video for all the themes? so that we could evaluate how that looks in all of them. That would be great!

@rhymes , @jacobherrington I updated the PR description with screenshots for all the themes. 

I noticed the color of the nav controls border ($dark-purple) doesn't change between themes, and that is also the color i used for the search focus border currently in this PR. Imho it looks good, please have a look when you get the chance and let me know if i should make this themable or not, or if there are any other suggestions/etc. Thank you!

**Comment by rhymes** [comment]
I tested it on local and I like it, though it's a bit "strong" in the pink mode.

I wonder if we shouldn't add some border shadow and lighthen the color up a bit though.

Do you think this should be themed @benhalpern?

**Pull request merged**